### ISSUE TITLE
[BUG] best_quality does not produce results better than medium_quality

### ISSUE THREAD

**Issue by aianthonyaz** [autogluon/autogluon#4410]
**Bug Report Checklist**

- [ X] I provided code that demonstrates a minimal reproducible example.  
- [ X] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [ X] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

First, let me say thank you!  This is a great tool which I have been using for a few months.  Thanks for all your efforts.

I have provided code,  and output which I have run on one of the latest versions: 1.1.2b20240816.  I can send you the csv files if you want them.  I would not allow me to attach them.

At the end of the .out file includes the output and has the classification report from all 4 quality levels.

 **best_quality** does not produce results better than **medium_quality**

For my application, I am trying to get the best quality positive case F1 score.  I don't care if it take 2 weeks to fit the model I want the most accurate model possible.  If there are any additional suggestions on getting the best accurate results, I would appreciate it.

Again, thank you for all you do with autogluon.  

**Expected behavior**
I would expect the quality to get better with each level of quality used. (medium, good, high, best)

**To Reproduce**
Run the python provided

**Screenshots / Logs**
attached

**Installed Versions**

INSTALLED VERSIONS
------------------
date                   : 2024-08-19
time                   : 12:08:03.176675
python                 : 3.10.14.final.0
OS                     : Linux
OS-release             : 6.8.0-40-generic
Version                : #40-Ubuntu SMP PREEMPT_DYNAMIC Fri Jul  5 10:34:03 UTC 2024
machine                : x86_64
processor              : x86_64
num_cores              : 12
cpu_ram_mb             : 15914.0390625
cuda version           : 12.535.183.01
num_gpus               : 1
gpu_ram_mb             : [5503]
avail_disk_size_mb     : 626098

accelerate             : 0.21.0
autogluon              : 1.1.2b20240816
autogluon.common       : 1.1.2b20240816
autogluon.core         : 1.1.2b20240816
autogluon.features     : 1.1.2b20240816
autogluon.multimodal   : 1.1.2b20240816
autogluon.tabular      : 1.1.2b20240816
autogluon.timeseries   : 1.1.2b20240816
boto3                  : 1.34.162
catboost               : 1.2.5
defusedxml             : 0.7.1
evaluate               : 0.4.2
fastai                 : 2.7.16
fugue                  : 0.9.2.dev1
gluonts                : 0.15.1
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.3
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.3.0
lightning              : 2.2.5
matplotlib             : 3.9.2
mlforecast             : 0.10.0
networkx               : 3.3
nlpaug                 : 1.1.11
nltk                   : 3.8.1
nptyping               : 2.4.1
numpy                  : 1.24.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : 1.18.1
optimum-intel          : None
orjson                 : 3.10.3
pandas                 : 2.2.2
pdf2image              : 1.17.0
Pillow                 : 10.2.0
psutil                 : 5.9.8
pytesseract            : 0.3.10
pytorch-lightning      : 2.3.3
pytorch-metric-learning: 2.3.0
ray                    : 2.31.0
requests               : 2.32.3
scikit-image           : 0.20.0
scikit-learn           : 1.3.2
scikit-learn-intelex   : None
scipy                  : 1.8.1
seqeval                : 1.2.2
setuptools             : 60.2.0
skl2onnx               : None
statsforecast          : 1.7.6
tabpfn                 : None
tensorboard            : 2.16.2
text-unidecode         : 1.3
timm                   : 0.9.16
torch                  : 2.2.0+cpu
torchmetrics           : 1.2.1
torchvision            : 0.17.0+cpu
tqdm                   : 4.65.2
transformers           : 4.39.3
utilsforecast          : 0.1.9
vowpalwabbit           : None
xgboost                : 1.7.1

</details>
[train.csv](https://github.com/user-attachments/files/16665161/train.csv)
[test.csv](https://github.com/user-attachments/files/16665162/test.csv)

**Comment by aianthonyaz** [comment]
[example.txt](https://github.com/user-attachments/files/16666327/example.txt)

**Comment by aianthonyaz** [comment]
Here is the python:

from autogluon.tabular import TabularDataset, TabularPredictor
import pandas as pd
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix, roc_curve, auc, precision_recall_curve, average_precision_score, precision_score, recall_score, jaccard_score, f1_score
from autogluon.core.utils import show_versions

show_versions()

presets = [['medium_quality', 'optimize_for_deployment'],
           ['good_quality', 'optimize_for_deployment'],
           ['high_quality', 'optimize_for_deployment'],
           ['best_quality', 'optimize_for_deployment']]

presets = [['medium_quality'],
           ['good_quality' ], 
           ['high_quality' ],
           ['best_quality' ]]

#presets = [['medium_quality', 'optimize_for_deployment'],
#           ['medium_quality', 'optimize_for_deployment']]

train = pd.read_csv("train.csv")
test = pd.read_csv("test.csv")
y_test = pd.read_csv('y_test.csv')

train_data = TabularDataset(train)
test_data = TabularDataset(test)

crs = []

for p in presets :
        predictor = TabularPredictor(label='TARGET').fit(train_data=train_data, presets=p, auto_stack=True)

        distilled_model_names = predictor.distill()
#        print(predictor.leaderboard(test_data))
        model_to_deploy = distilled_model_names[0]

        predictions = predictor.predict(test_data, model=model_to_deploy)

#        y_pred_proba = predictor.predict_proba(test_data)
#        y_pred_from_proba = predictor.predict_from_proba(y_pred_proba=y_pred_proba)


        print("****************************************")
        print(classification_report(y_test, predictions))
        print("****************************************")
        crs.append(classification_report(y_test, predictions))
        del predictor

for p in presets :
        print (p)
        print(crs.pop(0))

**Comment by Innixma** [comment]
If you care about F1 score, then tell AutoGluon to maximize F1 score and calibrate decision threshold:

```
TabularPredictor(..., eval_metric="f1").fit(..., calibrate_decision_threshold=True)
```
